Test/Graph/test_graph.py

```diff
@@ -235,8 +275,8 @@ def test_graph_wrong():
 
 
 def test_edges_are_correct():
-    check_edges(1, 1, False)
-    check_edges(1, 2, False)
+    # check_edges(1, 1, False)
+    # check_edges(1, 2, False) # KDTree removes the single site
```

Comment:
I am okay with having these fail (and, IMHO, the cases _with_ PBC and only one site should fail too; periodic boundaries for one site make even less sense unless we allow multiple edges and/or self loops, which we don't).
But there should be a better error message and the test should check that these cases error with `with pytest.raises(...): `.